addons/spectrum/functions/fnc_toggleJammingOnUnit.sqf

```diff
@@ -86,6 +88,15 @@ if (_enableJam) then {
 				missionNamespace setVariable [QGVAR(activeJammedUnits), _activelyJammedUnits, true];
 			};
 
+			// disconnect any player that might control this drone
+			private _droneUsers = (UAVControl _unit) select { !(_x isEqualType "String") };	 // leave only player objects
+			{
+				_x connectTerminalToUAV objNull; // disconnect player from drone
+				if (hasInterface) then {
```

Comment:
> This event is only raised on the server, so `hasInterface` would always return false, as you are checking the server, not the player.

Oh it is even worse than that. 😅 
When being a Spectrum Device jammer yourself and playing as both client and server you actually see this message every time you jam any drone (no matter if you are controlling it), because `hasInterface` is then actually `true`.
I made it a little less bad by d2f1d6782ac250eee34064eb505bade94cc08ffe.

> You could use Zeus Enhanceds hint event to send the hint as a `targetEvent` on the player. probably the easiest solution and avoids remoteExec. https://github.com/zen-mod/ZEN/blob/13fb2466794d64ac6cfac3e26bda36d9efcbd152/addons/common/XEH_postInit.sqf#L31C1-L34C33

I have to look into that. 🙂 
Dealing with such locality issues is still giving me a hard time if I don't just align with what you already did.

Reply:
You can do:
```sqf
// _message is used by ZEN as: hint _message
["zen_common_hint", [_message], _targetPlayer] call CBA_fnc_targetEvent;
```
And it will send an hint only to the player given with ``_targetPlayer``, so passing in the players object there would do the trick. 

``player`` will always return your "local" player in MP. So using ``player`` on a client, will return that clients player. So if you do ``player in _droneUsers`` on the server, as you are doing in  https://github.com/Crowdedlight/Crows-Electronic-Warfare/commit/d2f1d6782ac250eee34064eb505bade94cc08ffe then you are checking if the "servers" player, is present in that array and then showing hint to the server. Which won't work as on dedicated servers, server has no interface, and I actually don't know what ``player`` will return if on a dedicated server 😅 

I would suggest doing:
```sqf
private _droneUsers = (UAVControl _unit) select { !(_x isEqualType "String") };	 // leave only player objects
{
	_x connectTerminalToUAV objNull; // disconnect player from drone
	["zen_common_hint", ["Drone is jammed<br/><t color='#ff0000'>Connection lost</t>"], _x] call CBA_fnc_targetEvent; // hint to player why they got disconnected
} forEach _droneUsers;
```

As this sends an single event to the eventhandler with id "zen_common_hint" on the targets client, so on the players client. And there the Eventhandler I linked from ZEN then executes:
```sqf
    params ["_message"];
    hint _message;
```

Reply:
Fixed in 8f9e256cb023d4400cea79eab064a200b4719b7c
(I just needed to add `parseText` to your code to make it work.)